data_steward/cdr_cleaner/cleaning_rules/identifying_field_suppression.py

```diff
@@ -102,11 +103,12 @@ def get_query_specs(self, *args, **keyword_args) -> query_spec_list:
             are optional but the query is required.
         """
         queries = []
+        schemas = {}
+        for table in self.affected_tables:
+            schemas[table] = get_table_schema(table)
         for table in self.affected_tables:
-            bq_client = BigQueryClient(self.project_id)
-            schema = bq_client.get_table_schema(table)
```

Comment:
Thanks for bring that up so I can clarify. If you check my previous commit, I explored that option but realized that the list of `self.affected_tables` gets updated in between so some tables lack schema by the time get_query_specs is called. In the current commit, the list of `self.affected_tables` is the final updated list. You are describing my previous commit here which shows a failed test due to this difference in the list of affected tables.

Reply:
I agree that you don't need the `client` object. I don't think you need to use either the deprecated `get_table_schema` or create a new one either.  I would replace this with `fields_for`.  You stated you could not do that because

>  fields_for does not fulfill the requirement

What requirement is not fulfilled?  `fields_for` is used by ` get_table_schema` to access the JSON data.  I ran the following in a python interpreter and it worked as expected for me.
```python
import resources
schema = resources.fields_for('observation')
for item in schema:
    print(f"{item.get('mode')}\t\t{item.get('type')}\t\t{item.get('name')}")
```

Reply:
The following logic uses schema as a BigQuery `SchemaField` object list but `fields_for` returns a dictionary which can have idiosyncrasies and possibly change the existing logic. Since the CR works as-is except for the client side-effect, I was looking to avoid logic changes as a de facto requirement. However, the use case here is simple enough that `fields_for` could be used without logic changes, with existing checks for casing, so I'll make that change.

Reply:
@lrwb-aou Updated with `fields_for` instead of `get_table_schema`